### PR TITLE
:seedling: add GitHub Action to lint yaml

### LINKED ISSUE #3019: Add an action for linting the YAML files that are generated during scaffolding
### What do you want to happen?

**Description:**
We should add a GitHub Action to run lint checks against the YAML files that are generated as part of scaffolding out a project. 

A popular action seems to be: https://github.com/marketplace/actions/yaml-lint

**Motivation:**
Operator-SDK uses a `yamllint` check as part of the ansible plugin e2e molecule tests and it has failed a few times in the past due to trailing spaces or improper indenting from the YAML files generated by the kustomize plugin. In the past we have gone back and created PRs to fix the scaffolded files, but this becomes tedious to have to do every time we bump the Kubebuilder version we use.

It would be much nicer to have a validation check on Kubebuilder that ensures that the generate YAML passes lint checks that runs on all PRs so we can catch it earlier.

An example of a workaround the ansible plugin has had to implement due to a lint error from the YAML generated by Kubebuilder v3.7.0: https://github.com/operator-framework/operator-sdk/pull/6047#discussion_r995764399

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by everettraven**
## Description
- Add a GitHub Action job to run `yamllint` on the testdata directory. 
    - I used the same `yamllint` configuration that is used in the Operator-SDK ansible operator e2e tests: https://github.com/operator-framework/operator-sdk/blob/f298f7c92ee154a0e8123fb13398a7f21720cf0e/testdata/ansible/memcached-operator/molecule/default/molecule.yml#L8
- Fix any existing errors from running `yamllint -f parsable testdata --no-warnings -d "{extends: relaxed, rules: {line-length: {max: 120}}}"`

## Motivation
- Fixes #3019

### CODE REVIEW

**Review by Kavinjsir — COMMENTED** [review]
How about we have a separate yamllint config file?

Assuming this may help us keep lint style consist between remote local.

(Personally met more errors when running yamllint locally..)

```
❯ yamllint -d "{extends: relaxed, rules: {line-length: {max: 120}}}" --no-warnings testdata
testdata/project-v3/config/manager/manager.yaml
  40:73     error    trailing spaces  (trailing-spaces)

testdata/project-v3/config/certmanager/kustomizeconfig.yaml
  1:91      error    trailing spaces  (trailing-spaces)
...
```

**Comment by everettraven** [comment]
> How about we have a separate yamllint config file?
>Assuming this may help us keep lint style consist between remote and local.

@Kavinjsir Good point! I'll do that.

> (Personally met more errors when running yamllint locally..)

Did you run it on this PR branch or another branch? I included linting fixes in this PR, so if it wasn't on this PR branch then I would expect you to get errors.

**Comment by camilamacedo86** [comment]
Hi @everettraven,

What about we have a make target to allow test locally ? We can add this one in the lint check wdyt?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -50,3 +50,13 @@ jobs:
...
+
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: yaml-lint
+        uses: ibiqlik/action-yamllint@v3
+        with:
+          file_or_dir: testdata
+          config_data: "{extends: relaxed, rules: {line-length: {max: 120}}}"
```
Comment (camilamacedo86) [inline .github/workflows/lint.yml:62]: Nit: missing space

**Comment by Kavinjsir** [comment]
> > How about we have a separate yamllint config file?
> > Assuming this may help us keep lint style consist between remote and local.
> 
> @Kavinjsir Good point! I'll do that.
> 
> > (Personally met more errors when running yamllint locally..)
> 
> Did you run it on this PR branch or another branch? I included linting fixes in this PR, so if it wasn't on this PR branch then I would expect you to get errors.

@everettraven Thx for pointing out! I switched to your branch and it works for me (no more errors) 👍🏼

**Comment by everettraven** [comment]
@Kavinjsir @camilamacedo86 I have updated this PR to now add a `yamllint` target to the `Makefile` and run it as a prerequisite to the `lint` target.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -50,3 +50,10 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           labels: kind/bug
           body: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run yamllint make target
+        run: make yamllint
```
Comment (camilamacedo86) [inline .github/workflows/lint.yml:59]: nit space

**Review by camilamacedo86 — APPROVED** [review]
That is great for me 👍 

/approved

**Review by Kavinjsir — APPROVED** [review]
Looks good to me!

**Comment by everettraven** [comment]
@varshaprasad96 or @camilamacedo86 would either of you mind adding an LGTM so this PR will automerge?

**Review by varshaprasad96 — APPROVED** [review]
/lgtm

**Pull request merged**